apps/server/src/modules/tool/controller/dto/request/lti11-tool-config.params.ts

```diff
@@ -0,0 +1,30 @@
+import { ExternalToolConfigCreateParams } from '@src/modules/tool/controller/dto/request/external-tool-config.params';
+import { LtiMessageType } from '@src/modules/tool/interface/lti-message-type.enum';
+import { IsEnum, IsOptional, IsString } from 'class-validator';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+
+export class Lti11ToolConfigParams extends ExternalToolConfigCreateParams {
+	@IsString()
+	@ApiProperty()
+	type!: string;
+
+	@IsString()
+	@ApiProperty()
+	baseUrl!: string;
+
+	@IsString()
+	@ApiProperty()
+	key!: string;
+
+	@IsString()
+	@ApiProperty()
+	secret!: string;
+
+	@IsString()
+	@ApiProperty()
+	resource_link_id!: string;
```

Comment:
This is optional and gets filled with the course id, if not set. See https://github.com/hpi-schul-cloud/schulcloud-server/blob/edba8fb36f520a262b8549f4d43d1da2bf87b0e9/apps/server/src/modules/tool/uc/lti11.uc.ts#L36